lib/matplotlib/_constrained_layout.py

```diff
@@ -334,15 +428,16 @@ def _match_submerged_margins(fig):
     See test_constrained_layout::test_constrained_layout12 for an example.
     """
 
-    for panel in fig.subfigs:
-        _match_submerged_margins(panel)
+    for sfig in fig.subfigs:
+        match_submerged_margins(layoutgrids, sfig)
 
     axs = [a for a in fig.get_axes() if (hasattr(a, 'get_subplotspec')
                                          and a.get_in_layout())]
 
     for ax1 in axs:
         ss1 = ax1.get_subplotspec()
-        lg1 = ss1.get_gridspec()._layoutgrid
+        lg1 = (layoutgrids[ss1.get_gridspec()]
+               if ss1.get_gridspec() in layoutgrids else None)
         if lg1 is None:
             axs.remove(ax1)
             continue
```

Comment:
I _think_ `lg1` could never be `None` if it's in `layoutgrids`
```suggestion
        if ss1.get_gridspec() not in layoutgrids:
            axs.remove(ax1)
            continue
        lg1 = layoutgrids[ss1.get_gridspec()]
```
or:
```suggestion
        try:
            lg1 = layoutgrids[ss1.get_gridspec()]
        except KeyError:
            axs.remove(ax1)
            continue
```